reactos/dll/win32/browseui/basebarsite.cpp

```diff
@@ -475,7 +718,150 @@ LRESULT CBaseBarSite::OnNotify(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bH
     return 0;
 }
 
-HRESULT CreateBaseBarSite(REFIID riid, void **ppv)
+LRESULT CBaseBarSite::OnCommand(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    HRESULT                             hResult;
+    CComPtr<IDockingWindow>             parentSite;
+
+    if (HIWORD(wParam) == BN_CLICKED && LOWORD(wParam) == IDM_BASEBAR_CLOSE)
+    {
+        hResult = fDeskBarSite->QueryInterface(IID_PPV_ARG(IDockingWindow, &parentSite));
+        if (!SUCCEEDED(hResult))
+        {
+            return E_FAIL;
+        }
+        parentSite->ShowDW(FALSE);
+        bHandled = TRUE;
+    }
+    return 0;
+}
+
+LRESULT CBaseBarSite::OnCustomDraw(LPNMCUSTOMDRAW pnmcd)
+{
+    switch (pnmcd->dwDrawStage)
+    {
+        case CDDS_PREPAINT:
+        case CDDS_PREERASE:
+            return CDRF_NOTIFYITEMDRAW;
+        case CDDS_ITEMPREPAINT:
+            if (fVertical)
+            {
+                REBARBANDINFO info;
+                WCHAR wszTitle[MAX_PATH];
+                DWORD index;
+                RECT rt;
+                HFONT newFont;
+
+                index = SendMessage(RB_IDTOINDEX, fCurrentActiveBar->fBandID , 0);
+                ZeroMemory(&info, sizeof(info));
+                ZeroMemory(wszTitle, sizeof(wszTitle));
+                DrawEdge(pnmcd->hdc, &pnmcd->rc, EDGE_ETCHED, BF_BOTTOM);
+                // We also resize our close button
+                ::SetWindowPos(toolbarWnd, HWND_TOP, pnmcd->rc.right - 22, 0, 20, 18, SWP_SHOWWINDOW);
+                // Draw the text
+                info.cch = MAX_PATH;
+                info.lpText = wszTitle;
+                rt = pnmcd->rc;
+                rt.right -= 24;
+                rt.left += 2;
+                if(FAILED_UNEXPECTEDLY(GetInternalBandInfo(index, &info, RBBIM_TEXT)))
+                    return CDRF_SKIPDEFAULT;
+                newFont = GetTitleFont();
+                if(newFont)
+                    SelectObject(pnmcd->hdc, newFont);
+                DrawText(pnmcd->hdc, info.lpText, -1, &rt, DT_SINGLELINE | DT_LEFT);
+                DeleteObject(newFont);
+                return CDRF_SKIPDEFAULT;
+            }
+            else
+            {
+                DrawEdge(pnmcd->hdc, &pnmcd->rc, EDGE_ETCHED, BF_BOTTOM);
+                // We also resize our close button
+                ::SetWindowPos(toolbarWnd, HWND_TOP, 0, 2, 20, 18, SWP_SHOWWINDOW);
+            }
+            return CDRF_SKIPDEFAULT;
+        default:
+            break;
+    }
+    return CDRF_DODEFAULT;
+}
+
+HFONT CBaseBarSite::GetTitleFont()
+{
+    NONCLIENTMETRICS mt;
+    mt.cbSize = sizeof(mt);
+    if(!SystemParametersInfo(SPI_GETNONCLIENTMETRICS, sizeof(mt), &mt, 0))
+    {
+        ERR("Can't get system parameters !\n");
+        return NULL;
+    }
+    return CreateFontIndirect(&mt.lfMenuFont);
```

Comment:
Well, I'm too lazy to write code that reloads the font in case of theming change :D (even if our cabinet window doesn't forward those messages to child controls yet)